userComponent/src/main/kotlin/com/gravatar/app/usercomponent/data/InMemoryUserSessionPersistence.kt

```diff
@@ -0,0 +1,40 @@
+package com.gravatar.app.usercomponent.data
+
+import com.gravatar.app.foundations.DispatcherProvider
+import com.gravatar.app.usercomponent.domain.model.UserSession
+import com.gravatar.app.usercomponent.domain.repository.AuthRepository
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.launch
+
+internal class InMemoryUserSessionPersistence(
+    authRepository: AuthRepository,
+    applicationScope: CoroutineScope,
+    dispatcherProvider: DispatcherProvider,
+) : UserSessionPersistence {
+
+    private val _state: MutableSharedFlow<UserSession> = MutableSharedFlow(replay = 1)
+    override val state: Flow<UserSession> = _state
+
+    init {
+        applicationScope.launch(dispatcherProvider.io) {
```

Comment:
In order to initialize the state, we have to read from the DataStore, which is a suspending operation, so we need a scope. I've created an Application scope for that.